### PR TITLE
Update PR template to include the contributing doc; suggestion to not use force push

### PR DESCRIPTION
**Pull request by jihoonson**
This PR has 2 changes:

- Promoting [the contributing doc](https://github.com/apache/druid/blob/master/CONTRIBUTING.md). This change also includes a suggestion for not using force pushes and rebase after creating a PR.
- Rearrange the key changes of the PR. It seems reasonable to appear before the self-checklist.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -48,11 +57,4 @@ This PR has:
...
 
-<!-- Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the items which are not done or not relevant to the PR. None of the items from the checklist above are strictly necessary, but it would be very helpful if you at least self-review the PR. -->
-
-<hr>
-
-##### Key changed/added classes in this PR
- * `MyFoo`
- * `OurBar`
- * `TheirBaz`
+<!-- Once you open a PR, please _avoid using force pushes and rebasing_ since these make it difficult for reviewers to see what you've changed in response to their reviews. See [the contributing doc](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#if-your-pull-request-shows-conflicts-with-master) for more details. -->
```
Comment (capistrant) [inline .github/pull_request_template.md:60]: do we want to consider raising this comment up to the top of the template in the hopes of more people reading it vs it being at the bottom and possibly being missed/ignored?

Comment (jihoonson) [reply .github/pull_request_template.md]: Sounds good. I moved it to the top.

**Review by capistrant — APPROVED** [review]
Totally forgot about this, but I do still think it is an improvement and worth merging. +1 (I restarted flaky TravisCI job to try to get checks green)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -37,6 +39,15 @@ In each section, please describe design decisions made, including:
...
 <hr>
 
+##### Key changed/added classes in this PR
+ * `MyFoo`
+ * `OurBar`
+ * `TheirBaz`
+
+<hr>
+
+<!-- Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the items which are not done or not relevant to the PR. None of the items from the checklist above are strictly necessary, but it would be very helpful if you at least self-review the PR. -->
```
Comment (capistrant) [inline .github/pull_request_template.md:49]: nit: comment references "checklist above", but with this change the checklist is now below

Comment (jihoonson) [reply .github/pull_request_template.md]: Thanks, fixed it.

**Comment by jihoonson** [comment]
@capistrant thanks for the review. I will merge soon unless there is someone else wants to review.

**Pull request merged**